### DISCUSSION TITLE
Best practices when using Appender

### DISCUSSION THREAD

**Discussion by charlespnh** [duckdb/duckdb#13299] [Q&A]
I have an use case to handle traffic data and since it'll be write-heavy the Appender API is suitable for this. The traffic data will be parsed and inserted into 3 tables, so I'll have to create 3 appenders. I have a few questions regarding the best practices of using the Appender API in my particular cases:

1. Should each of this appender be created with a different connection, or can they share the same connection?
2. I'm storing these appenders as part of the application's lifetime, is this what I should be doing? Or is the appender only supposed to be short-lived in a function? It's a write-heavy use case so that's why I chose to go with the former, and the latter might incur overhead with creating a new connection and appender...

Thanks!

**Comment by chu-k** [comment]
Another appender-related question: How should an 'autoincrement' (sequence col) be handled when using an appender, as it seems a value would need to be explicitly provided for that column?

**Reply by Tishj** [reply]
1.1.0 will have AppendDefault, see <https://github.com/duckdb/duckdb/pull/11905>
No value needs to be explicitly provided if the column has a DEFAULT, then AppendDefault can be called instead

**Reply by chu-k** [reply]
Hi @Tishj, I was testing out AppendDefault on v1.1.0 today and encountered the unimplemented error with an autoincrement column as described above. Looking a bit closer at the [tests](https://github.com/Tishj/duckdb/blob/783826d7ded5eb6df05d612f2c0fc7e629c3ac69/test/appender/test_appender.cpp#L223) it looks like this case isn't supported in this version. Is there planned support for this or perhaps do you have suggestions on how I could contribute to implement?

**Comment by floatdrop** [comment]
> Should each of this appender be created with a different connection, or can they share the same connection?

I have tested multiple appenders on one connection (but for different tables) and it is working.

> I'm storing these appenders as part of the application's lifetime, is this what I should be doing?

Be aware, that appender will delay committing data to WAL before internal buffer is filled or appender is closed/flushed. It is good practice to call `Flush` method from time to time (for example with [debounce](https://github.com/floatdrop/debounce) in each AppendRow call).
